tests/auto/rt_auto.py

```diff
@@ -0,0 +1,474 @@
+# HEADER
+# Written by Brian Curtis
+# Automation of RT tasks using github cli
+from __future__ import print_function
+
+from github import Github as gh
+import datetime
+import time
+import pandas as pd
+import socket
+import threading
+import subprocess
+import re
+import sys
+import yaml
+import os
+import logging
+
+class RTData:
+
+    def __init__(self):
+        self.logger = logging.getLogger("RTData")
+        self.DB_FILENAME = 'rt_auto.yml'
+        self.data = self.read_yaml_data()
+
+    def read_yaml_data(self):
+        self.logger.info(f'Reading in default YAML data')
+        stream = open(self.DB_FILENAME, 'r')
+        yaml_data = yaml.load(stream, Loader=yaml.SafeLoader)
+        stream.close()
+        self.logger.info(f'Finished reading in YAML data')
+        return yaml_data
+
+    def get_yaml_subset(self, keyin):
+        self.logger.info(f'Getting YAML subset: {keyin}')
+        try:
+            yaml_subset = self.data[keyin]
+        except:
+            self.logger.critical(f'Unable to get yaml subset: {keyin}. Quitting')
+            sys.exit()
+        df = pd.DataFrame.from_dict(yaml_subset)
+        self.logger.info(f'Finished getting YAML subset {keyin}')
+        return df
+
+class Machine:
+
+    def __init__(self, rtdata_obj):
+        self.logger = logging.getLogger("Machine")
+        self.rtdata_obj = rtdata_obj
+        self.get_machine_info()
+        self.machineid = os.environ.get('MACHINE_ID')
+
+    def get_machine_info(self):
+        self.logger.info(f'Getting machine information')
+        hostname = socket.gethostname()
+        self.logger.debug(f'hostname is {hostname}')
+        machines = self.rtdata_obj.get_yaml_subset('machines')
+        regexmachines = machines['regexhostname'].values
+        for i, mach in enumerate(regexmachines):
+            if re.search(mach, hostname):
+                self.logger.info(f'{hostname} is an approved machine')
+                mach_db_info = machines.iloc[i]
+                self.name = mach_db_info['name']
+                self.regexhostname = mach_db_info['regexhostname']
+                self.workdir = mach_db_info['workdir']
+                self.baselinedir = mach_db_info['baselinedir']
+                self.logger.debug(f'self.name: {self.name}\n\
+                    self.regexhostname: {self.regexhostname}\n\
+                    self.workdir: {self.workdir}\n\
+                    self.baselinedir: {self.baselinedir}')
+                self.logger.info(f'Finished getting machine information')
+                return
+            else:
+                continue
+        self.logger.critical(f'Hostname:{hostname} does not match approved list. Quitting')
+        sys.exit()
+
+class Function:
```

Comment:
Function is too generic to named as a python Class.  I suggest renaming to "Action" or something of the sort.